src/preprocess/run_tg.py

```diff
@@ -43,105 +66,143 @@
     #           # scatter_plot_path=os.path.join(TG_OUT, 'global_pca.html'),
     #           bin_file_type='--bfile')
 
+
     # 1. Centralised variant QC
-    logger.info("Centralised variant QC")
     varqc_prefix = TG_BFILE_PATH + '_varqc'
-    QC.qc(input_prefix=TG_BFILE_PATH, output_prefix=varqc_prefix, qc_config=variant_qc_config)
+    if Stage.VARIANT_QC in stages:
+        logger.info('Centralised variant QC')
+        QC.qc(input_prefix=TG_BFILE_PATH, output_prefix=varqc_prefix, qc_config=variant_qc_config)
 
     # 2. Split into ethnic datasets and then QC each local dataset
-    logger.info("Splitting ethnic dataset")
-    nodes = SplitTGHeter().split(input_prefix=varqc_prefix, make_pgen=True)
+    nodes = set(TG_SUPERPOP_DICT.values())
+    if Stage.POPULATION_SPLIT in stages:
+        logger.info('Splitting ethnic dataset')
+        SplitTGHeter().split(input_prefix=varqc_prefix, make_pgen=True)
```

Comment:
Receive `nodes` list from `SplitHeter` class instead of `TG_SUPERPOP_DICT`, because won't work for homogeneous split.
`node` list will now include `ALL` node as it previously did.
```suggestion
        nodes = SplitTGHeter().split(input_prefix=varqc_prefix, make_pgen=True)
```

Reply:
Maybe then remove ALL from `.split` return list? Since ALL is not actually a node, it's kind of an identifier for various centralized things like PCA.

Reply:
It kinda always was there and removing it might potentially break something. And `ALL` is a node, it has folders  like other nodes do. I'm not sure if it is a good idea to go without it, also because `ALL` only_phenotypes folder would not be created, or is it?

Reply:
I think it was added by a mistake. We don't need to split data for ALL directory as we do it for all other nodes, because (1) it does not make sense; (2) it should not be used in other parts of the analysis. Data inside the ALL directory should correspond to the combined split structure for all other nodes that is not the case if we treat it as a usual node.

Reply:
But I see the problem with the homogeneous split. Let me refactor this.

Reply:
It does make sense, at least it did. Before pruning was added it `ALL` simulated centralized data. Only pruning step needs this `node_noall` list, because centralized model aggregates all individual nodes results. And centralized model existed separately before. Are they any other steps that require nodes aggregation?